### PR TITLE
fix Issue 17428 - [scope] class field assignment allows to escape sco…

### PR DESCRIPTION
**Pull request by WalterBright**
…pe parameters

### CODE REVIEW

File: test/fail_compilation/retscope2.d [hunk 1]
```diff
@@ -162,3 +162,46 @@ void foo800()
...
+
+
+
+
+
+
+
+
+
+/*************************************************/
```
Comment (mathias-lang-sociomantic) [inline test/fail_compilation/retscope2.d]: No need for this since you use `#line`.

File: test/fail_compilation/retscope2.d [hunk 2]
```diff
@@ -162,3 +162,46 @@ void foo800()
...
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope2.d(1005): Error: scope variable `p` assigned to `this` with longer lifetime
+---
+*/
+
+#line 1000
+
+class C17428
```
Comment (mathias-lang-sociomantic) [inline test/fail_compilation/retscope2.d]: Could you add a test with a `scope class` ?

Comment (WalterBright) [inline test/fail_compilation/retscope2.d]: That's probably already covered in the many tests in `retscope.d` and `retscope2.d`, but in any case it is off topic to this particular bug fix.

Comment (rainers) [inline test/fail_compilation/retscope2.d]: This compiles without complaint before this PR (which seems correct):
```
class C17428
{
    int* _p;
}

void main() @safe
{
	int x;
	int* p = &x;
	scope C17428 c;
	c._p = p;
}
```
but does no longer afterwards.

Comment (WalterBright) [inline test/fail_compilation/retscope2.d]: fixed

Comment (rainers) [inline test/fail_compilation/retscope2.d]: It now allows the assignment to `c._p` even without `scope`.

Comment (WalterBright) [inline test/fail_compilation/retscope2.d]: fixed

**Review by rainers — APPROVED** [review]

**Pull request merged**